cni/pkg/cmd/root.go

```diff
@@ -132,25 +132,25 @@ var rootCmd = &cobra.Command{
 			// if it is, we do NOT remove the plugin, and do
 			// NOT do ambient watch server cleanup
 			defer func() {
-				var isUpgrade bool
+				var shouldStopCleanup bool
 				if cfg.InstallConfig.AmbientDisableSafeUpgrade {
```

Comment:
Should we block cleanup if safeUpgrade is disabled but istio owned CNI config is enabled?

Reply:
Good point, maybe have `shouldStopCleanup` return a second boolean, `isUpgrade`, or something like that. Then we can try to make a more accurate decision about whether we think we should cleanup when safe upgrade is disabled?